agents/go-agents/src/main/go/core/httputils/httputils.go

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (c) 2012-2017 Codenvy, S.A.
+// All rights reserved. This program and the accompanying materials
+// are made available under the terms of the Eclipse Public License v1.0
+// which accompanies this distribution, and is available at
+// http://www.eclipse.org/legal/epl-v10.html
+//
+// Contributors:
+//   Codenvy, S.A. - initial API and implementation
+//
+
+package httputils
+
+import (
+	"net/http"
+	"regexp"
+)
+
+// BasePathChopper cuts base path of a request
+type BasePathChopper struct {
```

Comment:
This is kind of workaround for the `httprouter` implementation.
The previous (gorilla mux) based router could deal with regexp in paths, which didn't require such workaround, 
i think it makes sense to make this code `Router` implementation specific